File: src/main/java/com/pebble/baseAuth/config/oauth2/OAuth2UserInfoFactory.java
package com.pebble.baseAuth.config.oauth2;

import java.util.Map;

public class OAuth2UserInfoFactory {

    public static OAuth2UserInfo getOAuth2UserInfo(String registrationId, Map<String, Object> attributes) {
        if (registrationId.equalsIgnoreCase("google")) {
            return new GoogleOAuth2UserInfo(attributes);
        } else if (registrationId.equalsIgnoreCase("github")) {
            return new GitHubOAuth2UserInfo(attributes);
        } else {
            throw new IllegalArgumentException("지원하지 않는 소셜 로그인 제공자입니다: " + registrationId);
        }
    }
}

File: auth-module/src/main/kotlin/com/pebble/basicAuth/config/oauth2/OAuth2UserInfoFactory.kt
package com.pebble.basicAuth.config.oauth2

object OAuth2UserInfoFactory {
    @JvmStatic
    fun getOAuth2UserInfo(registrationId: String, attributes: Map<String, Any>): OAuth2UserInfo {
        return when (registrationId.lowercase()) {
            "google" -> GoogleOAuth2UserInfo(attributes)
            else -> throw IllegalArgumentException("ì§€?í•˜ì§€ ?ŠëŠ” ?Œì…œ ë¡œê·¸???œê³µ?ì…?ˆë‹¤: $registrationId")
        }
    }
}
